### PR TITLE
Fix bomb run slamming into mountainsides

### LINKED ISSUE #931: Planes crash into ground on high altitude airstrike
*Version:* 2.2.2

*Mods:* 

*Environment*: SP, MP LAN, MP host, MP dedi.

*.rpt attatched?* NO

*have you edited the missionfile?*: No

***Issue:***
due to how the air units are asked to fly on ingress to airstrike, they will crash into mountains if you call in a high altitude (on mountain or tall hill) they will slam into the ground as they cannot climb fast enough
---

### PR DESCRIPTION
**Pull request by HakonRydland**
## What type of PR is this.
1. [x] Bug
2. [ ] Enhancement

### What have you changed and why?
Information: made airstrike and NATObomb no go lower than their respective flyInHeight on bomb run target (init for NATObomb, marker for airstrike)
    

### Please specify which Issue this PR Resolves.
closes #931

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the Mission in Singleplayer?
2. [ ] Have you loaded the Mission in a Dedicated Server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

********************************************************
Notes:

### CODE REVIEW

**Comment by HakonRydland** [comment]
only last commit is the fix, the rest are git being bad, please discard these, only airstrike and NATObomb should be merged.

File: A3-Antistasi/functions/AI/fn_airstrike.sqf [hunk 1]
```diff
@@ -12,6 +12,9 @@ if (_markerX isEqualType "") then
 	_isMarker = true;
 	_positionX = getMarkerPos _markerX;
 	};
+_temp = "#lightpoint" createVehicleLocal _positionX;
+_posmrk = (getPosASL _temp);
+deleteVehicle _temp;
```
Comment (jaj22) [inline A3-Antistasi/functions/AI/fn_airstrike.sqf:17]: Doesn't this do the same thing as `_posmrk = ATLtoASL _positionX`?

Also _posmrk should be defined as private, moved closer to the only place that it's used, and probably given a more specific name. Given that you only need the z value, maybe something like:
```
private _minAltASL = ((ATLtoASL _positionX) select 2) + 150;
_plane flyInHeightASL [_minAltASL, _minAltASL, _minAltASL];
```

Concept looks correct.

Comment (HakonRydland) [reply A3-Antistasi/functions/AI/fn_airstrike.sqf]: reason for this method was that ATLtoASL returned unreliable results on repeated strikes on same target with no variable changed, chosen method resulted in constant same result on repeated strikes

Comment (jaj22) [reply A3-Antistasi/functions/AI/fn_airstrike.sqf]: The following test code returned identical results for _z1 and _z2 over a few hundred runs, so I can only guess that your test had other unrelated issues:

```
_xmod = (random 6000) - 3000;
_ymod = (random 6000) - 3000;
_pos = position player vectorAdd [_xmod, _ymod, 0];

_z1 = (ATLtoASL _pos) select 2;

_temp = "#lightpoint" createVehicleLocal _pos;
_z2 = (getPosASL _temp) select 2; 
+deleteVehicle _temp;

[_z1, _z2, _pos]; 
```

**Comment by Spoffy** [comment]
Manually merged in ae9dbbc1c10b59162699bd2382fcba82affad866 - And Jaj is totally right, the ATLtoASL approach is the way to go, as 'createVehicle' is pretty damn expensive in terms of time!

**Pull request closed without merging**